src/coreclr/jit/gtlist.h

```diff
@@ -248,10 +246,19 @@ GTNODE(SELECTCC         , GenTreeOpCC        ,0,GTK_BINOP|DBK_NOTHIR)
 // sets the condition flags to the specified immediate value.
 GTNODE(CCMP             , GenTreeCCMP        ,0,GTK_BINOP|GTK_NOVALUE|DBK_NOTHIR)
 // Maps to arm64 cinc instruction. It returns the operand incremented by one when the condition is true.
-// Otherwise returns the unchanged operand. Optimises for patterns such as, result = condition ? op1 + 1 : op1
-GTNODE(CINC             , GenTreeOp          ,0,GTK_BINOP|DBK_NOTHIR)
-// Variant of CINC that reuses flags computed by a previous node with the specified condition.
-GTNODE(CINCCC           , GenTreeOpCC        ,0,GTK_UNOP|DBK_NOTHIR)
+// Otherwise returns the unchanged operand. Optimises for patterns such as, result = condition ? op1 : op1 + 1
+GTNODE(SELECT_INC             , GenTreeOp          ,0,GTK_BINOP|DBK_NOTHIR)
+// Variant of SELECT_INC that reuses flags computed by a previous node with the specified condition.
+GTNODE(SELECT_INCCC           , GenTreeOpCC        ,0,GTK_BINOP|DBK_NOTHIR)
+// Maps to arm64 csinv instruction. It negates the operand when the condition is true.
+// Otherwise returns the unchanged operand. Optimises for patterns such as, result = condition ? op1 : ~op2
+GTNODE(SELECT_INV             , GenTreeOp          ,0,GTK_BINOP|DBK_NOTHIR)
+// Variant of SELECT_INV that reuses flags computed by a previous node with the specified condition.
+GTNODE(SELECT_INVCC           , GenTreeOpCC        ,0,GTK_BINOP|DBK_NOTHIR)
+// Conditional select and negate when conditiona is false. Optimises for patterns such as, result = condition ? -op1 : op2
+GTNODE(SELECT_NEG             , GenTreeOp          ,0,GTK_BINOP|DBK_NOTHIR)
+// Variant of SELECT_NEG that reuses flags computed by a previous node with the specified condition.
+GTNODE(SELECT_NEGCC           , GenTreeOpCC        ,0,GTK_BINOP|DBK_NOTHIR)
```

Comment:
I think the descriptions need to be updated to clarify that both one operand and two operand versions are supported. And I think describing the semantics in code is much simpler to understand (instead of "Optimises for patterns such as" just call this out as being the semantics of this oper).

E.g.
"If op2 is null, computes result = condition ? op1 : op1 + 1. Otherwise computes result = condition ? op1 : op2 + 1".

Reply:
Done